### PR TITLE
Add option for hiding profile from all trees

### PR DESCRIPTION
**Pull request by rudyflores**
## Proposed changes

Solves #2567 

Add ability for the user to hide profiles in the tree in sync or individually based on their choice in the quickpick

## Sample

https://github.com/zowe/vscode-extension-for-zowe/assets/68666202/8fa5f69b-7d24-4f54-ae50-8734dba58272

## Release Notes

Milestone: 2.13.0

Changelog: Add support for hiding a Zowe profile across all trees

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Review by zFernand0 — CHANGES_REQUESTED** [review]
The prompt and everything around the functionality LGTM! 😋 

I did notice that some of the old options came back in the USS context menu.
image
This doesn't happen (for me) on 2.12.1.

Also, the Edit history is above the Manage profile on the Jobs (but this also happens on the main branch)

**Comment by rudyflores** [comment]
> The prompt and everything around the functionality LGTM! 😋
> 
> I did notice that some of the old options came back in the USS context menu. image This doesn't happen (for me) on 2.12.1.
> 
> Also, the Edit history is above the Manage profile on the Jobs (but this also happens on the main branch)

Hey Fernando! That issue should be resolved once this PR is merged https://github.com/zowe/vscode-extension-for-zowe/pull/2559

As for the menu order issue you mentioned that is not relevant to this change, but could create an issue for that an I can address that separately :)

File: packages/zowe-explorer/src/uss/init.ts [hunk 1]
```diff
@@ -191,5 +191,5 @@ export async function initUSSProvider(context: vscode.ExtensionContext): Promise
     );
 
     initSubscribers(context, ussFileProvider);
-    return ussFileProvider;
+    return ussFileProvider as USSTree;
```
Comment (traeok) [inline packages/zowe-explorer/src/uss/init.ts:194]: We can avoid this type cast by removing the type annotation or changing the annotation to `USSTree` on line 27.

**Review by traeok — DISMISSED** [review]
LGTM - left a comment about a type cast, but the current logic won't affect the behavior 😅 
Thanks @rudyflores!

**Review by zFernand0 — COMMENTED** [review]
LGTM! 😋 
I'll wait for the PR below before approving
- #2559

**Review by traeok — APPROVED** [review]
Thanks again for removing the cast and fixing the errors from the type change 😁

**Review by JillieBeanSim — COMMENTED** [review]
Hey @rudyflores this is looking great! I do have one question.
If the profile name is only in a single tree, is there a way for us to know this and react appropriately?  I tested this situation and had a profile name in a single tree and was still prompted to choose.  I didn't get any errors or anything from choosing all trees, but wonder if we can tell beforehand, and if only in a single tree just remove it instead of the prompt that doesn't really make sense if only in one tree.

**Comment by rudyflores** [comment]
> Hey @rudyflores this is looking great! I do have one question. If the profile name is only in a single tree, is there a way for us to know this and react appropriately? I tested this situation and had a profile name in a single tree and was still prompted to choose. I didn't get any errors or anything from choosing all trees, but wonder if we can tell beforehand, and if only in a single tree just remove it instead of the prompt that doesn't really make sense if only in one tree.

Perhaps we can cycle through each tree and see if the session exists in atleast another tree, if only in the current tree we can just remove the current one?

**Review by traeok — APPROVED** [review]

**Review by zFernand0 — APPROVED** [review]
I was testing right before the meeting, 
I just forgot to approve :cry:

LGTM! :yum:

**Pull request merged**